File: demo-custom-annotation-processor/app/src/main/java/be/lennertsoffers/demo_custom_annotation_processor/MainActivity.java
package be.lennertsoffers.demo_custom_annotation_processor;

import android.os.Bundle;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;
import androidx.navigation.NavController;
import androidx.navigation.fragment.NavHostFragment;
import androidx.navigation.ui.NavigationUI;

import be.lennertsoffers.annotations_library.Configure;
import be.lennertsoffers.demo_custom_annotation_processor.fragments.HomeFragmentDirections;


@Configure(packageLocation = "be.lennertsoffers.demo_custom_annotation_processor", databaseName = "test_db", databaseVersion = 1)
public class MainActivity extends AppCompatActivity {
    public static be.lennertsoffers.demo_custom_annotation_processor.EasyPersistenceDatabase db;
    private NavController navController;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        NavHostFragment navHostFragment = (NavHostFragment) this.getSupportFragmentManager().findFragmentById(R.id.fragmentContainer);
        if (navHostFragment != null) this.navController = navHostFragment.getNavController();
        NavigationUI.setupActionBarWithNavController(this, this.navController);

        db = new be.lennertsoffers.demo_custom_annotation_processor.EasyPersistenceDatabase(this.getApplicationContext());
    }

    @Override
    protected void onStop() {
        db.close();
        super.onStop();
    }

    public void handleShowCarsClick(View view) {
        this.navController.navigate(HomeFragmentDirections.actionHomeFragmentToCarsFragment());
    }

    public void handleShowPersonsClick(View view) {
        this.navController.navigate(HomeFragmentDirections.actionHomeFragmentToPersonsFragment());
    }
}

File: demo-room-annotation-processor/app/src/main/java/be/lennertsoffers/demo_room_annotation_processor/MainActivity.kt
package be.lennertsoffers.demo_room_annotation_processor

import androidx.navigation.ui.NavigationUI.setupActionBarWithNavController
import androidx.appcompat.app.AppCompatActivity
import androidx.navigation.NavController
import android.os.Bundle
import androidx.navigation.fragment.NavHostFragment
import androidx.room.Room
import be.lennertsoffers.demo_room_annotation_processor.fragments.HomeFragmentDirections

class MainActivity : AppCompatActivity() {
    private var navController: NavController? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val navHostFragment = this.supportFragmentManager.findFragmentById(R.id.fragmentContainer) as NavHostFragment?
        if (navHostFragment != null) navController = navHostFragment.navController
        setupActionBarWithNavController(this, navController!!)

        db = Room.databaseBuilder(
            applicationContext,
            DemoRoomDatabase::class.java,
            "room_demo_db"
        ).allowMainThreadQueries().build()
    }

    override fun onStop() {
        db.close()
        super.onStop()
    }

    fun handleShowCarsClick() {
        navController!!.navigate(HomeFragmentDirections.actionHomeFragmentToCarsFragment())
    }

    fun handleShowPersonsClick() {
        navController!!.navigate(HomeFragmentDirections.actionHomeFragmentToPersonsFragment())
    }

    companion object {
        private lateinit var db: DemoRoomDatabase

        fun getDb(): DemoRoomDatabase {
            return this.db
        }
    }
}